### PR TITLE
MEPTS-675: New Report - MISAU Key Population

### PR DESCRIPTION
**Pull request by amosh101**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/MISAUKeyPopsCohortQueries.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+import org.openmrs.module.reporting.cohort.definition.CompositionCohortDefinition;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/** Defines all of the TxCurrCohortQueries we want to expose for EPTS */
+@Component
+public class MISAUKeyPopsCohortQueries {
+
+  @Autowired private ResumoMensalCohortQueries resumoMensalCohortQueries;
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/MISAUKeyPopsCohortQueries.java]: Now this being a new class, can we adopt the bean creation through a constructor with only one @Autowired place at the method signature? this the recommended srping way, reduces on time for initial start up and can be called directitly in the tests classes

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/MISAUKeyPopsDataSetDefinition.java [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MISAUKeyPopsDataSetDefinition extends BaseDataSet {
+
+  @Autowired private MISAUKeyPopsCohortQueries mISAUKeyPopsCohortQueries;
+
+  @Autowired private EptsGeneralIndicator eptsGeneralIndicator;
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/MISAUKeyPopsDataSetDefinition.java]: Apply the same technique here by populating those in a constructor

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupMISAUKeyPopsReport.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+@Component
+public class SetupMISAUKeyPopsReport extends EptsDataExportManager {
+
+  private MISAUKeyPopsDataSetDefinition mISAUKeyPopsDataSetDefinition;
+
+  private GenericCohortQueries genericCohortQueries;
+
+  @Autowired
+  public SetupMISAUKeyPopsReport(
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupMISAUKeyPopsReport.java:41]: Good stuff, this what am referring to in the other comments, please replicate uniformity

**Review by ningosi — COMMENTED** [review]
@amosh101  your work is great, a few alignments to adjust there

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/MISAUKeyPopsCohortQueries.java (hunk 1 above)
Comment (amosh101) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/MISAUKeyPopsCohortQueries.java]: Thanks Nicholas, please check now.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupMISAUKeyPopsReport.java (hunk 3 above)
Comment (amosh101) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupMISAUKeyPopsReport.java:41]: Understood.

**Review by ningosi — APPROVED** [review]

**Pull request merged**